Kaniko integration tests failing - likely related to Github Actions worker update

Currently all of Kaniko's integration tests are failing @ HEAD with the below log snippet.  Doesn't seem related to a direct change in Kaniko code but likely a change in the Github Action worker image (dep update). Need to root cause and fix this issue:

log link:
https://github.com/GoogleContainerTools/kaniko/actions/runs/6489137282/job/17622851856?pr=2791

log error snippet
```
integration_test.go:604: Error building image: Failed to build image localhost:5000/kaniko-dockerfile_test_cache with kaniko command "[docker run --net=host -e BENCHMARK_FILE=false -v /home/runner/work/kaniko/kaniko/integration:/workspace -v /tmp/2024572772:/kaniko/benchmarks -v /home/runner/.docker/config.json:/root/.docker/config.json -e DOCKER_CONFIG=/root/.docker executor-image -f /workspace/dockerfiles/Dockerfile_test_cache -d localhost:5000/kaniko-dockerfile_test_cache --force -c /workspace]": exit status 1
            error checking push permissions -- make sure you entered the correct tag name, and that you are authenticated correctly, and try again: checking push permission for "localhost:5000/kaniko-dockerfile_test_cache": creating push check transport for localhost:5000 failed: Get "https://localhost:5000/v2/": dial tcp [::1]:5000: connect: connection refused; Get "http://localhost:5000/v2/": dial tcp [::1]:5000: connect: connection refused
```

--- a/.github/workflows/integration-tests.yaml
+++ b/.github/workflows/integration-tests.yaml
@@ -26,6 +26,12 @@ jobs:
         - k8s-executor-build-push integration-test-k8s
 
     steps:
+    - name: Maximize build space
+      uses: AdityaGarg8/remove-unwanted-software@v1
+      with:
+        remove-android: 'true'
+        remove-dotnet: 'true'
+        remove-haskell: 'true'
     - uses: actions/setup-go@93397bea11091df50f3d7e59dc26a7711a8bcfbe # v4.1.0
       with:
         go-version: '1.20'

